### PR TITLE
task 1 done

### PR DESCRIPTION
**Pull request by ladamashina**

### CODE REVIEW

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -238,7 +283,12 @@ function encodeToRot13(str) {
...
  */
 function isString(value) {
-    throw new Error('Not implemented');
+   if ( typeof(value) === "string" || value instanceof String ){
+    return  true
+   }
+    else {
+        return false;
+    }
 }
```
Comment (IlyaMokin) [inline task/01-strings-tasks.js]: "if else"  is redundant here

File: task/01-strings-tasks.js [hunk 2]
```diff
@@ -114,7 +125,14 @@ function removeLeadingAndTrailingWhitespaces(value) {
...
-    throw new Error('Not implemented');
+    //throw new Error('Not implemented');
+
+     var result = value;
+    for(var a=1; a<count; a++){
+        result = result + value;
+    }
+    return result;
+    
 }
```
Comment (IlyaMokin) [inline task/01-strings-tasks.js]: https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/String/repeat

File: task/01-strings-tasks.js [hunk 3]
```diff
@@ -55,7 +57,8 @@ function getStringLength(value) {
  *   'Chuck','Norris'  => 'Hello, Chuck Norris!'
  */
 function getStringFromTemplate(firstName, lastName) {
-    throw new Error('Not implemented');
+   // throw new Error('Not implemented');
+   return ('Hello, ' + firstName + ' '+ lastName+'!')
 }
```
Comment (IlyaMokin) [inline task/01-strings-tasks.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/template_strings

File: task/04-arrays-tasks.js [hunk 4]
```diff
@@ -37,9 +37,17 @@ function findElement(arr, value) {
  *    2 => [ 1, 3 ] 
  *    5 => [ 1, 3, 5, 7, 9 ]
  */
-function generateOdds(len) {
-   throw new Error('Not implemented');
-}
+function generateOdds(len) {     
+    var array = new Array(len) ;
+    array.fill(0);
```
Comment (AlexMokin) [inline task/04-arrays-tasks.js]: what does fill return? can you chain fill and map?

File: task/04-arrays-tasks.js [hunk 5]
```diff
@@ -234,7 +265,12 @@ function toArrayOfSquares(arr) {
  *   [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ] => [ 1, 3, 6, 10, 15, 21, 28, 36, 45, 55 ]
  */
 function getMovingSum(arr) {
-   throw new Error('Not implemented');
+    var sum=0;
+    arr=arr.map(function(el,index){ 
+        sum = sum + el;
```
Comment (AlexMokin) [inline task/04-arrays-tasks.js]: see reduce method. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce?v=b

**Comment by IlyaMokin** [comment]
done

**Pull request closed without merging**